sharing-service/services/sharing/share.js

```diff
@@ -0,0 +1,114 @@
+const { createSignedLink } = require('./lib/signcode')
+const { getUserIdFromEmail } = require('slic-tools/get')
+
+const signedAxios = require('aws-signed-axios')
+const awsXray = require('aws-xray-sdk')
+const AWS = require('aws-sdk')
+
+const log = require('slic-tools/log')
+
+const SQS = awsXray.captureAWSClient(
+  new AWS.SQS({ endpoint: process.env.SQS_ENDPOINT_URL })
+)
+const SSM = awsXray.captureAWSClient(
+  new AWS.SSM({ endpoint: process.env.SSM_ENDPOINT_URL })
+)
+
+const queueName = process.env.EMAIL_QUEUE_NAME
+if (!queueName) {
+  throw new Error('EMAIL_QUEUE_NAME must be set')
+} else {
+  log.info({ queueName }, 'Using queue')
+}
+
+const queueUrlPromise = fetchQueueUrl()
+queueUrlPromise
+  .then(() => {
+    log.info('promise handled - queueUrlPromise')
+  })
+  .catch(err => {
+    log.info('promise rejected - queueUrlPromise', err)
+    console.dir(err)
+  })
+
+async function fetchQueueUrl() {
+  const queueUrl = (await SQS.getQueueUrl({
+    QueueName: 'dev-email-queue'
+  }).promise()).QueueUrl
+  log.info({ queueUrl }, 'Using queue URL')
+
+  return queueUrl
+}
+
+const userServiceUrlPromise = getUserServiceUrl()
+userServiceUrlPromise
+  .then(() => {
+    log.info('userServiceUrlPromise resolved')
+  })
+  .catch(() => {
+    log.info('userServiceUrlPromise not resolved: failed')
+  })
+async function getUserServiceUrl() {
+  const result = await SSM.getParameter({ Name: 'UserServiceUrl' }).promise()
+  const {
+    Parameter: { Value: userServiceUrl }
+  } = result
+  return userServiceUrl
+}
+async function create({ email, listId }) {
+  let link
+  let message
+  const nsDomain = process.env.SLIC_NS_DOMAIN
+
+  if(!nsDomain){
+    log.info('SLIC_NS_DOMAIN must be set')
+  }
+
+  log.info('attributes in share.create: ', email, listId)
```

Comment:
Use correct syntax for logging consistent with other services, if necessary. If the logging statements are not required for normal operation, remove the log statement or use `log.debug`

```
log.debug({ email, listId }, 'Creating a share')
```